modules/aniviewBidAdapter.js

```diff
@@ -1,317 +1,321 @@
-          s2sParams.AV_GDPR = 1;
-          s2sParams.AV_CONSENT = bidderRequest.gdprConsent.consentString;
-        }
-      }
-      if (bidderRequest && bidderRequest.uspConsent) {
-        s2sParams.AV_CCPA = bidderRequest.uspConsent;
-      }
+  isBidRequestValid(bid) {
+    return !!(bid.params?.AV_PUBLISHERID && bid.params?.AV_CHANNELID);
+  },
+
+  buildRequests(bidRequests, bidderRequest) {
+    const requests = [];
 
-      let serverDomain = bidRequest.params && bidRequest.params.serverDomain ? bidRequest.params.serverDomain : 'gov.aniview.com';
-      let servingUrl = 'https://' + serverDomain + '/api/adserver/vast3/';
+    bidRequests.forEach((bidRequest) => {
+      Object.keys(bidRequest.mediaTypes).forEach((mediaType) => {
+        const endpoint = bidRequest.params.dev?.endpoint || SSP_ENDPOINT;
 
-      bidRequests.push({
-        method: 'GET',
-        url: servingUrl,
-        data: s2sParams,
-        bidRequest
+        requests.push({
+          method: 'POST',
+          url: endpoint,
+          bids: [bidRequest],
+          options: { withCredentials: false },
+          data: converter.toORTB({
+            bidderRequest,
+            bidRequests: [bidRequest],
+            context: { mediaType },
+          }),
+        });
       });
+    });
+
+    return requests;
+  },
+
+  interpretResponse(serverResponse, bidderRequest) {
+    const { bids, data } = bidderRequest;
+    const { body } = serverResponse;
+    const bidResponse = body?.seatbid?.[0]?.bid?.[0];
+
+    if (!bids || !data || !bidResponse) {
+      return [];
     }
-  }
 
-  return bidRequests;
-}
-function getCpmData(xml) {
-  let ret = {cpm: 0, currency: 'USD'};
-  if (xml) {
-    let ext = xml.getElementsByTagName('Extensions');
-    if (ext && ext.length > 0) {
-      ext = ext[0].getElementsByTagName('Extension');
-      if (ext && ext.length > 0) {
-        for (var i = 0; i < ext.length; i++) {
-          if (ext[i].getAttribute('type') == 'ANIVIEW') {
-            let price = ext[i].getElementsByTagName('Cpm');
-            if (price && price.length == 1) {
-              ret.cpm = price[0].textContent;
-            }
-            break;
-          }
-        }
+    const response = converter.fromORTB({ response: body, request: data });
+    const bid = response.bids[0];
+    const replacements = {
+      auctionPrice: bid.cpm,
+      auctionId: bid.requestId,
+      auctionBidId: bidResponse.impid,
+      auctionImpId: bidResponse.impid,
+      auctionSeatId: bid.seatBidId,
+      auctionAdId: bid.adId,
+    };
+    const bidAdmWithReplacedMacros = replaceMacros(bidResponse.adm, replacements);
+
+    if (bid.mediaType === VIDEO) {
+      bid.vastXml = bidAdmWithReplacedMacros;
+
+      if (bidResponse?.nurl) {
+        bid.vastUrl = replaceMacros(bidResponse.nurl, replacements);
       }
+    } else {
+      bid.ad = bidAdmWithReplacedMacros;
     }
-  }
-  return ret;
+
+    return response.bids;
```

Comment:
shouldn't you return `[ bid ]` here ? 
It feels the `return response.bids` can only work because bid[0] would be changes by reference and because you expect to have only one bid response. 
Returning only the modified bid object seems to be more resilient